calico_containers/adapter/datastore.py

```diff
@@ -444,36 +463,112 @@ def get_bgp_peers(self, version):
         Get the configured BGP Peers
 
         :param version: "v4" for IPv4, "v6" for IPv6
-        :return: List of netaddr.IPAddress IP addresses.
+        :return: List of dictionaries:
+                    [
+                      {
+                        "ip": <peer ip address>,
+                        "as_num": <peer as num>
+                      },
+                      ...
+                    ]
         """
         assert version in ("v4", "v6")
-        bgp_peer_path = BGP_PEER_PATH % {"version": version}
-        return map(IPAddress, self._get_path_with_keys(bgp_peer_path).keys())
+        bgp_peers_path = BGP_PEERS_PATH % {"version": version}
 
-    def _get_path_with_keys(self, path):
+        try:
+            nodes = self.etcd_client.read(bgp_peers_path).children
+        except EtcdKeyNotFound:
+            # Path doesn't exist.
+            return []
+
+        peers = []
+        for node in nodes:
+            if node.value is None:
+                continue
+            try:
+                data = json.loads(node.value)
+                data["ip"] = IPAddress(data["ip"])
+                peers.append(data)
+            except BaseException:
```

Comment:
Pokemon exception handling?  (Gotta' catch 'em all.)

This seems very broad.  When would we hit an exception here?